.spectral.yaml

```
@@ -28,6 +28,34 @@ rules:
       field: example
       function: truthy
 
+  property-string-max-length:
+    description: Every string type should define a max length
+    message: "{{path}} does not define max length"
+    given: $..[?((@property=='items' || @parentProperty=='properties') && @.type == 'string')]
+    severity: error
+    then:
+      field: maxLength
+      function: truthy
+
+  property-field-sentence:
+    description: Description of a field should be a sentence
+    message: "{{path}} is not a sentence"
+    given: $..[?(@property=='description')]
+    severity: error
+    then:
+      function: pattern
+      functionOptions:
+        match: '^[A-Z]+\s*\S+(?:\s+\S+){2,}\s*$'
```

```suggestion
        match: '^[A-Z]*\s*\S+(?:\s+\S+){2,}\s*$'
```
In this case the expression `A valid expression` will fail because of `^[A-Z]+` because it requires one more character, to work properly you can replace `+` by `*`, in this case `+` means one or more while `*` means zero or more.